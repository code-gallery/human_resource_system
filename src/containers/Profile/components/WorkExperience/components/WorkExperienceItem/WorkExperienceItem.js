import React from 'react'
import PropTypes from 'prop-types'
import WorkItemView from '../WorkItemView'
import WorkItemEdit from '../WorkItemEdit'

class WorkExperienceItem extends React.Component {
  constructor(props) {
    super(props)
    this.state = {
      showDesc: false
    }
  }

  componentWillReceiveProps = (nextProps) => {
    if (this.state.edit && this.props.editMode && !nextProps.editMode) {
      this.toggleEdit()
    }
  }

  toggleEdit = () => {
    this.setState({
      edit: !this.state.edit
    })
    this.props.toggleBlockEditing()
  }

  render() {
    const { edit } = this.state
    const { deleteWork } = this.props
    return (
      <div>
        <WorkItemView
          {...this.props}
          toggleEdit={this.toggleEdit}
        />
        {
          edit &&
          <WorkItemEdit
            {...this.props}
            toggleEdit={this.toggleEdit}
            edit={true}
            deleteWork={deleteWork}
          />
        }
      </div>
    )
  }
}

WorkExperienceItem.propTypes = {
  toggleBlockEditing: PropTypes.func.isRequired,
  deleteWork: PropTypes.func.isRequired,
  saveEntity: PropTypes.func.isRequired,
  editMode: PropTypes.bool
}

export default WorkExperienceItem
