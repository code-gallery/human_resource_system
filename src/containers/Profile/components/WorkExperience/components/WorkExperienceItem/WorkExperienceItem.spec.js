import React from 'react'
import renderer from 'react-test-renderer'

import WorkExperienceItem from './index'
import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from 'enzyme';


describe('containers/Profile/components/.../WorkExperienceItem', () => {
  const commonProps = {
    start_date: '2017-08-01T00:00:00.000Z',
    end_date: '0000-00-00',
    name: 'Dev 2',
    verified_status: 'verified',
    org: {
      id: 55,
      name: 'Applied Blockchain',
      logo_image: '//logo_image'
    },
    industries: [
      { text: 'Accounting' },
      { text: 'Animation' },
      { text: 'IT' }
    ],
    toggleBlockEditing: jest.fn(),
    deleteWork: jest.fn(),
    saveEntity: jest.fn()
  }

  describe('when editMode is false', () => {
    it('renders correctly', () => {
      const props = {
        ...commonProps,
        editMode: false
      }
      const tree = renderer.create(
        <WorkExperienceItem {...props} />
      ).toJSON()

      expect(tree).toMatchSnapshot()
    })
  })

  describe('when editMode is true', () => {
    const props = {
      ...commonProps,
      editMode: true
    }

    it('renders correctly', () => {
      const tree = renderer.create(
        <WorkExperienceItem {...props} />
      ).toJSON()

      expect(tree).toMatchSnapshot()
    })
    configure({adapter: new Adapter()});
    describe('UNSAFE_componentWillReceiveProps', () => {
      it('calls props.toggleBlockEditing correctly', () => {
        const wrapper = shallow(
          <WorkExperienceItem {...props} />
        )
        expect(props.toggleBlockEditing).not.toHaveBeenCalled()
        wrapper.setState({ edit: true }, function() {
          wrapper.setProps({
            editMode: false
          })
        })
        expect(props.toggleBlockEditing).toHaveBeenCalled()
        expect(wrapper.state('edit')).toEqual(false)
      })
    })
  })
})
