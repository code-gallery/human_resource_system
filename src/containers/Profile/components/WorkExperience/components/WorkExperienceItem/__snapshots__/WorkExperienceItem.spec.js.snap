// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`containers/Profile/components/.../WorkExperienceItem when editMode is false renders correctly 1`] = `
<div>
  <div
    className="profile-experience-block profile-verified-status"
  >
    <div
      className="relative"
    >
      <div
        className="Avatar Avatar-white Avatar-size-109"
      >
        <img
          alt="logo"
          src="appii-placeholder.png"
        />
      </div>
      <p
        className="visible-xs visible-sm"
      >
        <i
          className="profile-status-ico tooltip-position-top tooltip-movable tooltip-success"
          data-tooltip="Verified"
        >
           
        </i>
      </p>
    </div>
    <div
      className="profile-experience-info"
    >
      <i
        className="profile-status-ico tooltip-position-top tooltip-movable tooltip-success hidden-xs hidden-sm"
        data-tooltip="Verified"
      >
         
      </i>
      <p>
        Applied Blockchain
      </p>
      <span />
      <em>
        2017
         - 
        Present
      </em>
    </div>
  </div>
</div>
`;

exports[`containers/Profile/components/.../WorkExperienceItem when editMode is true renders correctly 1`] = `
<div>
  <div
    className="profile-experience-block profile-verified-status"
  >
    <div
      className="profile-experience-btns"
    >
      <button
        className="edit-btn"
        onClick={[Function]}
      >
         
      </button>
    </div>
    <div
      className="relative"
    >
      <div
        className="Avatar Avatar-white Avatar-size-109"
      >
        <img
          alt="logo"
          src="appii-placeholder.png"
        />
      </div>
      <p
        className="visible-xs visible-sm"
      >
        <i
          className="profile-status-ico tooltip-position-top tooltip-movable tooltip-success"
          data-tooltip="Verified"
        >
           
        </i>
      </p>
    </div>
    <div
      className="profile-experience-info"
    >
      <i
        className="profile-status-ico tooltip-position-top tooltip-movable tooltip-success hidden-xs hidden-sm"
        data-tooltip="Verified"
      >
         
      </i>
      <p>
        Applied Blockchain
      </p>
      <span />
      <em>
        2017
         - 
        Present
      </em>
    </div>
  </div>
</div>
`;
